### PR TITLE
adaptations to new nix version, tiny changes to the cmake files

### PR DESCRIPTION
**Pull request by jgrewe**
This is an intermediate version, do not yet merge into master. But feel free to comment, checkout and try yourself :)

Builds successfully.
Tests, however crash at functions that test sources.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,10 +3,13 @@ project(nix_mx CXX)
 
 set(VERSION_MAJOR 1)
 set(VERSION_MINOR 4)
-set(VERSION_PATCH 0)
+set(VERSION_PATCH 2)
 
 set(VERSION_ABI   1)
 
+ENABLE_LANGUAGE(C)
+cmake_policy(SET CMP0050 NEW)
```
Comment (gicmo) [inline CMakeLists.txt]: I comment here would be useful

File: src/nixfile.cc [hunk 2]
```diff
@@ -28,9 +28,9 @@ mxArray *message(std::vector<nix::valid::Message> mes) {
 
         nix::valid::Message curr = mes[i];
 
-        struct_builder msb({ 1 }, { "id", "name", "msg" });
+        struct_builder msb({ 1 }, { "id", "msg" });
         msb.set(curr.id);
-        msb.set(curr.name);
+        //msb.set(curr.name);
```
Comment (gicmo) [inline src/nixfile.cc]: just delete it?

File: CMakeLists.txt (hunk 1 above)
Comment (jgrewe) [reply CMakeLists.txt]: haha, you found the weak point. I got an cmake error that language c was not enabled, so I did this. The cmake_policy warning could be dealt with by explicitly stating the policy. The policy changes do not apply to our cmake solution, afaik, so I chose it... If you have more insight into it, however, ...

**Comment by jgrewe** [comment]
@gicmo: 
Here's the crash report...
```
MATLAB crash file:/Users/jan/matlab_crash_dump.18615-1:


--------------------------------------------------------------------------------
        Segmentation violation detected at Di Okt 02 09:37:49 2018 +0200
--------------------------------------------------------------------------------

Fault Count: 1

Abnormal termination

Register State (from fault):
  RAX = 0000600001d933a0  RBX = 000060000150e880
  RCX = 00007fff7fcf5fdf  RDX = 000070000e178f08
  RSP = 000000018314be0c  RBP = 000070000e179050
  RSI = 000070000e178f08  RDI = 000070000e178f10

   R8 = 000000000000001d   R9 = 00006000049562a0
  R10 = 0000600004a4ba60  R11 = 000000000e178f20
  R12 = 000070000e179520  R13 = 000000018314b4bb
  R14 = 0000000000000002  R15 = 0000000000000000

  RIP = 000070000e179310  RFL = 000000018328da44

   CS = 000070000e179718   FS = 000000000000001d   GS = 000070000e179718

Stack Trace (from fault):
[  0] 0x0000000111ae3dd1                           bin/maci64/libmwfl.dylib+00007633 _ZN10foundation4core4diag15stacktrace_base7captureERKNS1_14thread_contextEm+00000049
[  1] 0x0000000111ae94b5                           bin/maci64/libmwfl.dylib+00029877 _ZN10foundation4core4test17terminate_handledEv+00006181
[  2] 0x0000000111ae79f5                           bin/maci64/libmwfl.dylib+00023029 _ZN10foundation4core4diag13terminate_logEPKcPK17__darwin_ucontext+00000149
[  3] 0x00000001166db67b                          bin/maci64/libmwmcr.dylib+00632443 _Z19mnPrintErrorMessageRKNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEE+00012635
[  4] 0x00000001166d9207                          bin/maci64/libmwmcr.dylib+00623111 _Z19mnPrintErrorMessageRKNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEE+00003303
[  5] 0x00000001166d6d64                          bin/maci64/libmwmcr.dylib+00613732 mnFatalSignalHandler+00000148
[  6] 0x00007fff7fd25b3d           /usr/lib/system/libsystem_platform.dylib+00019261 _sigtramp+00000029
[  7] 0x000070000e179318                                   <unknown-module>+00000000
[  8] 0x000000018328da44 /Users/jan/projects/programming/nix-mx/build/nix_mx.mexmaci64+01337924 _Z13mx_to_stringsPK11mxArray_tag+00000276
[  9] 0x00000001831bd39b /Users/jan/projects/programming/nix-mx/build/nix_mx.mexmaci64+00484251 _ZNK9extractor3vecINSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEEEENS1_6vectorIT_NS5_IS9_EEEEm+00000043
[ 10] 0x00000001831bd358 /Users/jan/projects/programming/nix-mx/build/nix_mx.mexmaci64+00484184 _ZN4glue5funky9ex_getterINSt3__16vectorINS2_12basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEENS7_IS9_EEEEvE3getERK9extractori+00000040
[ 11] 0x00000001831bd326 /Users/jan/projects/programming/nix-mx/build/nix_mx.mexmaci64+00484134 _ZN4glue5funky10matryoshkaIMN3nix9DataArrayEFNS2_12SetDimensionERKNSt3__16vectorINS5_12basic_stringIcNS5_11char_traitsIcEENS5_9allocatorIcEEEENSA_ISC_EEEEEE3getERK9extractori+00000038
[ 12] 0x00000001831bd2ea /Users/jan/projects/programming/nix-mx/build/nix_mx.mexmaci64+00484074 _ZN4glue5funky20matryoshka_unwrapperILi0ENSt3__16vectorINS2_12basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEENS7_IS9_EEEEE5valueIRNS0_10matryoshkaIMN3nix9DataArrayEFNSF_12SetDimensionERKSB_EEEEESB_OT_RK9extractori+00000042
[ 13] 0x00000001831bd23e /Users/jan/projects/programming/nix-mx/build/nix_mx.mexmaci64+00483902 _ZN4glue5funky14matryoshka_getILi0ERNS0_10matryoshkaIMN3nix9DataArrayEFNS3_12SetDimensionERKNSt3__16vectorINS6_12basic_stringIcNS6_11char_traitsIcEENS6_9allocatorIcEEEENSB_ISD_EEEEEEEEENS0_22matryoshka_return_typeIXT_ET0_E4typeERK9extractorOSN_+00000046
[ 14] 0x00000001831bd13e /Users/jan/projects/programming/nix-mx/build/nix_mx.mexmaci64+00483646 _ZN4glue5funky7invokerIN3nix9DataArrayEMS3_FNS2_12SetDimensionERKNSt3__16vectorINS5_12basic_stringIcNS5_11char_traitsIcEENS5_9allocatorIcEEEENSA_ISC_EEEEES4_E6invokeIRNS0_10matryoshkaISI_EEJLi0EEEEvSI_OT_RK9extractorR7infusorNS0_4iseqIJXspT0_EEEE+00000222
[ 15] 0x00000001831bd051 /Users/jan/projects/programming/nix-mx/build/nix_mx.mexmaci64+00483409 _ZN4glue5funky7funcboxIN3nix9DataArrayEMS3_FNS2_12SetDimensionERKNSt3__16vectorINS5_12basic_stringIcNS5_11char_traitsIcEENS5_9allocatorIcEEEENSA_ISC_EEEEEEclERK9extractorR7infusor+00000065
[ 16] 0x000000018314ab99 /Users/jan/projects/programming/nix-mx/build/nix_mx.mexmaci64+00015257 _ZN4glue8registry8dispatchERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEERK9extractorR7infusor+00000649
[ 17] 0x000000018314a663 /Users/jan/projects/programming/nix-mx/build/nix_mx.mexmaci64+00013923 mexFunction+00000499
[ 18] 0x00000001177614bc                            bin/maci64/libmex.dylib+00242876 _ZNK18C_DispatchStrategy8dispatchEiPP11mxArray_tagiS2_+00000140
[ 19] 0x0000000117761a6d                            bin/maci64/libmex.dylib+00244333 _ZNK24SCCompatDispatchStrategy8dispatchEiPP11mxArray_tagiS2_+00000061
[ 20] 0x0000000117761274                            bin/maci64/libmex.dylib+00242292 _ZN7Mfh_mex30dispatch_file_with_protectionsEiPP11mxArray_tagiS2_+00000484
[ 21] 0x0000000117748dbf                            bin/maci64/libmex.dylib+00142783 _ZN7Mfh_mex13dispatch_fileEiPP11mxArray_tagiS2_+00000159
[ 22] 0x000000011681c40a                 bin/maci64/libmwm_dispatcher.dylib+00500746 _ZN8Mfh_file16dispatch_fh_implEMS_FviPP11mxArray_tagiS2_EiS2_iS2_+00000746
[ 23] 0x000000011681c593                 bin/maci64/libmwm_dispatcher.dylib+00501139 _ZN8Mfh_file11dispatch_fhEiPP11mxArray_tagiS2_+00000035
[ 24] 0x00000001181befd7                        bin/maci64/libmwm_lxe.dylib+09338839 _ZN9MathWorks3lxe17LxeDispatcherImpl12apply_handleEPNS0_29DispatcherFunctionHandleTokenEmPNS0_6xvalueEmPKS4_NS0_9ApplyTypeE+00001303
[ 25] 0x00000001182be2eb                        bin/maci64/libmwm_lxe.dylib+10384107 _ZN9MathWorks3lxe12apply_handleEPvNS_2ts4TypeEPKvS1_S1_+00000507
[ 26] 0x000000011821ccde                        bin/maci64/libmwm_lxe.dylib+09723102 _ZN9MathWorks3lxe38cpp_primitive_template_fn_with_contextIXadL_ZNS0_12apply_handleEPvNS_2ts4TypeEPKvS2_S2_EEEEbS2_S4_S6_S2_S2_+00000014
[ 27] 0x0000000117a51740                        bin/maci64/libmwm_lxe.dylib+01550144 _ZN9MathWorks3lxe13IrInterpreter17ApplyCppPrimitiveEPN2ir4termES4_RKNS_2ts4TypeES8_+00000448
[ 28] 0x0000000117a50ad2                        bin/maci64/libmwm_lxe.dylib+01546962 _ZN9MathWorks3lxe13IrInterpreter31process_application_of_constantEPN2ir4termE+00001202
[ 29] 0x0000000117a4f32d                        bin/maci64/libmwm_lxe.dylib+01540909 _ZN9MathWorks3lxe13IrInterpreter13applyConstantEPN2ir4termE+00000077
[ 30] 0x0000000117a4ad13                        bin/maci64/libmwm_lxe.dylib+01522963 _ZN9MathWorks3lxe12IRTreeWalker9applyTermEPN2ir4termE+00000099
[ 31] 0x0000000117a4afb9                        bin/maci64/libmwm_lxe.dylib+01523641 _ZN9MathWorks3lxe12IRTreeWalker11getNextIRPCENS0_9irpc_typeE+00000217
[ 32] 0x0000000117a4f0c7                        bin/maci64/libmwm_lxe.dylib+01540295 _ZN9MathWorks3lxe13IrInterpreter9tree_walkERKNS0_9irpc_typeES4_+00000215
[ 33] 0x0000000117a4efab                        bin/maci64/libmwm_lxe.dylib+01540011 _ZN9MathWorks3lxe13IrInterpreter9InterpretEv+00000091
[ 34] 0x0000000117b5cfaf                        bin/maci64/libmwm_lxe.dylib+02645935 _ZN9MathWorks3lxe15ExecutionDriver9ThreadRunEv+00000911
[ 35] 0x0000000117b62107                        bin/maci64/libmwm_lxe.dylib+02666759 _ZN9MathWorks3lxe12IrEngineImpl5ApplyERKN2ir4termEPvS6_+00000295
[ 36] 0x000000011809a0b9                        bin/maci64/libmwm_lxe.dylib+08138937 _ZN9MathWorks3lxe17IrEngineDecorator5ApplyERKN2ir4termEPvS6_+00000153
[ 37] 0x00000001181b5bf9                        bin/maci64/libmwm_lxe.dylib+09300985 _ZN9MathWorks3lxe27FileBasedMatlabCodeFunction14DispatchScriptEiPP11mxArray_tagiS4_+00000297
[ 38] 0x00000001181b5e51                        bin/maci64/libmwm_lxe.dylib+09301585 _ZN9MathWorks3lxe27FileBasedMatlabCodeFunction12DispatchFileEiPP11mxArray_tagiS4_b+00000129
[ 39] 0x000000011681c40a                 bin/maci64/libmwm_dispatcher.dylib+00500746 _ZN8Mfh_file16dispatch_fh_implEMS_FviPP11mxArray_tagiS2_EiS2_iS2_+00000746
[ 40] 0x000000011681c593                 bin/maci64/libmwm_dispatcher.dylib+00501139 _ZN8Mfh_file11dispatch_fhEiPP11mxArray_tagiS2_+00000035
[ 41] 0x00000001181befd7                        bin/maci64/libmwm_lxe.dylib+09338839 _ZN9MathWorks3lxe17LxeDispatcherImpl12apply_handleEPNS0_29DispatcherFunctionHandleTokenEmPNS0_6xvalueEmPKS4_NS0_9ApplyTypeE+00001303
[ 42] 0x00000001182be989                        bin/maci64/libmwm_lxe.dylib+10385801 _ZN9MathWorks3lxe16ans_apply_handleEPvNS_2ts4TypeEPKvS1_S1_+00000361
[ 43] 0x000000011821cd5e                        bin/maci64/libmwm_lxe.dylib+09723230 _ZN9MathWorks3lxe38cpp_primitive_template_fn_with_contextIXadL_ZNS0_16ans_apply_handleEPvNS_2ts4TypeEPKvS2_S2_EEEEbS2_S4_S6_S2_S2_+00000014
[ 44] 0x0000000117a51740                        bin/maci64/libmwm_lxe.dylib+01550144 _ZN9MathWorks3lxe13IrInterpreter17ApplyCppPrimitiveEPN2ir4termES4_RKNS_2ts4TypeES8_+00000448
[ 45] 0x0000000117a50ad2                        bin/maci64/libmwm_lxe.dylib+01546962 _ZN9MathWorks3lxe13IrInterpreter31process_application_of_constantEPN2ir4termE+00001202
[ 46] 0x0000000117a4f32d                        bin/maci64/libmwm_lxe.dylib+01540909 _ZN9MathWorks3lxe13IrInterpreter13applyConstantEPN2ir4termE+00000077
[ 47] 0x0000000117a4ad13                        bin/maci64/libmwm_lxe.dylib+01522963 _ZN9MathWorks3lxe12IRTreeWalker9applyTermEPN2ir4termE+00000099
[ 48] 0x0000000117a4afb9                        bin/maci64/libmwm_lxe.dylib+01523641 _ZN9MathWorks3lxe12IRTreeWalker11getNextIRPCENS0_9irpc_typeE+00000217
[ 49] 0x0000000117a4f0c7                        bin/maci64/libmwm_lxe.dylib+01540295 _ZN9MathWorks3lxe13IrInterpreter9tree_walkERKNS0_9irpc_typeES4_+00000215
[ 50] 0x0000000117a4efab                        bin/maci64/libmwm_lxe.dylib+01540011 _ZN9MathWorks3lxe13IrInterpreter9InterpretEv+00000091
[ 51] 0x0000000117b5cfaf                        bin/maci64/libmwm_lxe.dylib+02645935 _ZN9MathWorks3lxe15ExecutionDriver9ThreadRunEv+00000911
[ 52] 0x0000000117b62107                        bin/maci64/libmwm_lxe.dylib+02666759 _ZN9MathWorks3lxe12IrEngineImpl5ApplyERKN2ir4termEPvS6_+00000295
[ 53] 0x000000011809a0b9                        bin/maci64/libmwm_lxe.dylib+08138937 _ZN9MathWorks3lxe17IrEngineDecorator5ApplyERKN2ir4termEPvS6_+00000153
[ 54] 0x00000001181b5bf9                        bin/maci64/libmwm_lxe.dylib+09300985 _ZN9MathWorks3lxe27FileBasedMatlabCodeFunction14DispatchScriptEiPP11mxArray_tagiS4_+00000297
[ 55] 0x00000001181b5e51                        bin/maci64/libmwm_lxe.dylib+09301585 _ZN9MathWorks3lxe27FileBasedMatlabCodeFunction12DispatchFileEiPP11mxArray_tagiS4_b+00000129
[ 56] 0x000000011681c40a                 bin/maci64/libmwm_dispatcher.dylib+00500746 _ZN8Mfh_file16dispatch_fh_implEMS_FviPP11mxArray_tagiS2_EiS2_iS2_+00000746
[ 57] 0x000000011681c593                 bin/maci64/libmwm_dispatcher.dylib+00501139 _ZN8Mfh_file11dispatch_fhEiPP11mxArray_tagiS2_+00000035
[ 58] 0x00000001181befd7                        bin/maci64/libmwm_lxe.dylib+09338839 _ZN9MathWorks3lxe17LxeDispatcherImpl12apply_handleEPNS0_29DispatcherFunctionHandleTokenEmPNS0_6xvalueEmPKS4_NS0_9ApplyTypeE+00001303
[ 59] 0x00000001182be989                        bin/maci64/libmwm_lxe.dylib+10385801 _ZN9MathWorks3lxe16ans_apply_handleEPvNS_2ts4TypeEPKvS1_S1_+00000361
[ 60] 0x000000011821cd5e                        bin/maci64/libmwm_lxe.dylib+09723230 _ZN9MathWorks3lxe38cpp_primitive_template_fn_with_contextIXadL_ZNS0_16ans_apply_handleEPvNS_2ts4TypeEPKvS2_S2_EEEEbS2_S4_S6_S2_S2_+00000014
[ 61] 0x0000000117a51740                        bin/maci64/libmwm_lxe.dylib+01550144 _ZN9MathWorks3lxe13IrInterpreter17ApplyCppPrimitiveEPN2ir4termES4_RKNS_2ts4TypeES8_+00000448
[ 62] 0x0000000117a50ad2                        bin/maci64/libmwm_lxe.dylib+01546962 _ZN9MathWorks3lxe13IrInterpreter31process_application_of_constantEPN2ir4termE+00001202
[ 63] 0x0000000117a4f32d                        bin/maci64/libmwm_lxe.dylib+01540909 _ZN9MathWorks3lxe13IrInterpreter13applyConstantEPN2ir4termE+00000077
[ 64] 0x0000000117a4ad13                        bin/maci64/libmwm_lxe.dylib+01522963 _ZN9MathWorks3lxe12IRTreeWalker9applyTermEPN2ir4termE+00000099
[ 65] 0x0000000117a4afb9                        bin/maci64/libmwm_lxe.dylib+01523641 _ZN9MathWorks3lxe12IRTreeWalker11getNextIRPCENS0_9irpc_typeE+00000217
[ 66] 0x0000000117a4f0c7                        bin/maci64/libmwm_lxe.dylib+01540295 _ZN9MathWorks3lxe13IrInterpreter9tree_walkERKNS0_9irpc_typeES4_+00000215
[ 67] 0x0000000117a4efab                        bin/maci64/libmwm_lxe.dylib+01540011 _ZN9MathWorks3lxe13IrInterpreter9InterpretEv+00000091
[ 68] 0x0000000117b5cfaf                        bin/maci64/libmwm_lxe.dylib+02645935 _ZN9MathWorks3lxe15ExecutionDriver9ThreadRunEv+00000911
[ 69] 0x0000000117b62107                        bin/maci64/libmwm_lxe.dylib+02666759 _ZN9MathWorks3lxe12IrEngineImpl5ApplyERKN2ir4termEPvS6_+00000295
[ 70] 0x000000011809a0b9                        bin/maci64/libmwm_lxe.dylib+08138937 _ZN9MathWorks3lxe17IrEngineDecorator5ApplyERKN2ir4termEPvS6_+00000153
[ 71] 0x0000000118049609                        bin/maci64/libmwm_lxe.dylib+07808521 _ZN9MathWorks3lxe9Evaluator15ApplyStatementsERKNSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEERKN2ir4termERNS0_18IEvaluationContextE+00000425
[ 72] 0x000000011804d0ee                        bin/maci64/libmwm_lxe.dylib+07823598 _ZN9MathWorks3lxe9Evaluator17EvalStatementImplERKNSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEERNS0_18IEvaluationContextE+00000206
[ 73] 0x000000011804d1ba                        bin/maci64/libmwm_lxe.dylib+07823802 _ZN9MathWorks3lxe9Evaluator13EvalStatementERKNSt3__112basic_stringIDsNS2_11char_traitsIDsEENS2_9allocatorIDsEEEERNS0_18IEvaluationContextE+00000106
[ 74] 0x00000001180d5b87                        bin/maci64/libmwm_lxe.dylib+08383367 _ZN9MathWorks3lxe9LxeBridge7EvalCmdERKNSt3__112basic_stringIDsNS2_11char_traitsIDsEENS2_9allocatorIDsEEEEb+00000327
[ 75] 0x00000001180d5a77                        bin/maci64/libmwm_lxe.dylib+08383095 _ZN9MathWorks3lxe9LxeBridge7EvalCmdERKNSt3__112basic_stringIDsNS2_11char_traitsIDsEENS2_9allocatorIDsEEEEb+00000055
[ 76] 0x000000011786cd8d                       bin/maci64/libmwbridge.dylib+00073101 _Z8mnParserv+00000653
[ 77] 0x00000001166708c9                          bin/maci64/libmwmcr.dylib+00194761 _ZN3mcr7runtime26InterpreterThreadSingleton4Impl16break_event_loopERKN7mwboost10shared_ptrIS2_EE+00003369
[ 78] 0x0000000116670bf9                          bin/maci64/libmwmcr.dylib+00195577 _ZN3mcr7runtime26InterpreterThreadSingleton4Impl16break_event_loopERKN7mwboost10shared_ptrIS2_EE+00004185
[ 79] 0x000000011664d1f5                          bin/maci64/libmwmcr.dylib+00049653 _ZN14cmddistributor15PackagedTaskIIP13getTypeStringIvEENSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEv+00000917
[ 80] 0x000000011664d137                          bin/maci64/libmwmcr.dylib+00049463 _ZN14cmddistributor15PackagedTaskIIP13getTypeStringIvEENSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEv+00000727
[ 81] 0x000000011664d0e2                          bin/maci64/libmwmcr.dylib+00049378 _ZN14cmddistributor15PackagedTaskIIP13getTypeStringIvEENSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEv+00000642
[ 82] 0x000000011654d74e                          bin/maci64/libmwiqm.dylib+00366414 _ZN7mwboost6detail8function21function_obj_invoker0INSt3__18functionIFNS_3anyEvEEES5_E6invokeERNS1_15function_bufferE+00000030
[ 83] 0x000000011654d16b                          bin/maci64/libmwiqm.dylib+00364907 _ZN3iqm18PackagedTaskPlugin7executeEP15inWorkSpace_tag+00000251
[ 84] 0x000000011652eb5f                          bin/maci64/libmwiqm.dylib+00240479 _ZNK3iqm17DefaultPacketInfo7executeEP15inWorkSpace_tag+00000799
[ 85] 0x000000011651299b                          bin/maci64/libmwiqm.dylib+00125339 _ZN3iqm10DefaultIqm13dequeueHelperEN7mwboost8optionalImEERKNSt3__112basic_stringIDsNS4_11char_traitsIDsEENS4_9allocatorIDsEEEEN6mlutil14cmddistributor13WhenToDequeue13WhenToDequeueERKNS4_8functionIFbRKNS4_5tupleIJSG_NS1_10shared_ptrINS_22IntermediatePacketInfoEEESA_EEEEEE+00001195
[ 86] 0x0000000116506dca                          bin/maci64/libmwiqm.dylib+00077258 _ZN3iqm10DefaultIqm7dequeueEN6mlutil14cmddistributor13WhenToDequeue13WhenToDequeueEN7mwboost8optionalImEE+00000362
[ 87] 0x0000000116518766                          bin/maci64/libmwiqm.dylib+00149350 _ZN3tbb10interface519concurrent_hash_mapIN6mlutil10contextmgr5McrIDEN7mwboost10shared_ptrIN3iqm17DebugEventHandlerEEEN10foundation7msg_svc10threadutil4HashIS4_EENS_13tbb_allocatorINSt3__14pairIS4_S9_EEEEE28allocate_node_copy_constructERNSF_INSK_4nodeEEERKS4_PKS9_+00006198
[ 88] 0x000000010f6e88e5                     bin/maci64/libmwservices.dylib+01239269 _Z40svWS_ProcessPendingEventsWithLimitedRateb+00003301
[ 89] 0x000000010f6e70f6                     bin/maci64/libmwservices.dylib+01233142 _Z25svWS_ProcessPendingEventsiib+00001094
[ 90] 0x000000010f6e6d5f                     bin/maci64/libmwservices.dylib+01232223 _Z25svWS_ProcessPendingEventsiib+00000175
[ 91] 0x000000011666f284                          bin/maci64/libmwmcr.dylib+00189060 _ZN3mcr7runtime26InterpreterThreadSingleton4Impl14process_eventsERKN7mwboost10shared_ptrIS2_EE+00000356
[ 92] 0x000000011666f957                          bin/maci64/libmwmcr.dylib+00190807 _ZN3mcr7runtime26InterpreterThreadSingleton4Impl3runERKN7mwboost10shared_ptrIS2_EEPNS2_12init_contextE+00000279
[ 93] 0x000000011666ac1a                          bin/maci64/libmwmcr.dylib+00171034 _Z26run_init_and_handle_eventsPv+00000266
[ 94] 0x00007fff7fd2e33d            /usr/lib/system/libsystem_pthread.dylib+00013117 _pthread_body+00000126
[ 95] 0x00007fff7fd312a7            /usr/lib/system/libsystem_pthread.dylib+00025255 _pthread_start+00000070
[ 96] 0x00007fff7fd2d425            /usr/lib/system/libsystem_pthread.dylib+00009253 thread_start+00000013
```

Can you get some sense out of it?

**Comment by jgrewe** [comment]
all problems resolved, all tests pass! The current bindings are hiding the optional/default arguments that were introduced in nix 1.4.2

**Pull request merged**